src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -154,6 +155,7 @@ public static Birthday parseBirthday(String birthday) throws ParseException {
      */
     public static IcNumber parseIcNumber(String icNumber) throws ParseException {
         requireNonNull(icNumber);
+        icNumber = icNumber.toUpperCase();
```

Comment:
Good job with these minor details eg (making some attributes like Gender and IC Number) case insensitive. This definitely accounts for unexpected scenarios of inputs while ensuring the system works as expected!